src/gg.regression.unity.bots/Editor/Scripts/RGInstrumentation.cs

```diff
@@ -323,92 +320,132 @@ private static bool ApplyCodeCovInstrumentation(ModuleDefinition module, MethodR
+#if ENABLE_LEGACY_INPUT_MANAGER
                     {
-                        if (ApplyCodeCovInstrumentation(module, visitMethod))
+                        var wrapperMethods = FindWrapperMethods(wrapperModule);
+                        if (ApplyLegacyInputInstrumentation(module, wrapperMethods))
                         {
                             anyChanges = true;
+                            RGDebug.LogInfo($"Instrumented legacy input API usage in assembly: {assemblyPath}");
                         }
                     }
-                    else
+#endif
+
+                    if (codeCovEnabled)
+                    {
+                        var visitMethod = FindCodeCoverageVisitMethod(wrapperModule);
+                        if (visitMethod != null)
+                        {
+                            if (ApplyCodeCovInstrumentation(module, visitMethod))
+                            {
+                                anyChanges = true;
+                            }
+                        }
+                        else
+                        {
+                            RGDebug.LogError($"Failed to find RGCodeCoverage.Visit method, code coverage instrumentation not applied");
+                        }
+                    }
+
+                    if (anyChanges)
                     {
-                        RGDebug.LogError($"Failed to find RGCodeCoverage.Visit method, code coverage instrumentation not applied");
+                        module.Write(tmpOutputPath, new WriterParameters()
+                        {
+                            WriteSymbols = true,
+                            SymbolWriterProvider = new PdbWriterProvider()
+                        });
                     }
                 }
 
                 if (anyChanges)
                 {
-                    module.Write(tmpOutputPath, new WriterParameters()
+                    string pdbPath = Path.ChangeExtension(assemblyPath, ".pdb");
+                    File.Delete(assemblyPath);
+                    if (File.Exists(pdbPath))
                     {
-                        WriteSymbols = true,
-                        SymbolWriterProvider = new PdbWriterProvider()
-                    });
+                        File.Delete(pdbPath);
+                    }
+
+                    string outPdbPath = Path.ChangeExtension(tmpOutputPath, ".pdb");
+                    File.Move(tmpOutputPath, assemblyPath);
+                    if (File.Exists(outPdbPath))
+                    {
+                        File.Move(outPdbPath, pdbPath);
+                    }
                 }
             }
+        }
+
+        private static readonly string RgAsmName = Path.GetFileName(typeof(BotSegmentsPlaybackController).Assembly.Location);
+
+        private static Assembly GetAssemblyForPath(string assemblyPath, out Assembly rgAssembly)
+        {
+            var assemblies = CompilationPipeline.GetAssemblies(AssembliesType.Player);
```

Comment:
Almost all the changes in this file are to minimize calls made to `CompilationPipeline.GetAssemblies`